### DISCUSSION TITLE
Score function "IV-type" for PLIV with two IVs

### DISCUSSION THREAD

**Discussion by AdamBaumann** [DoubleML/doubleml-for-r#182] [Q&A] [answered]
When trying to compare the result of using the "partialling out" and "IV-type" score functions for PLIV with two IVs there is seemingly only the option of using "partialling out". Is it possible to implement the IV-type score function here in a similar fashion as for the PLIV with one IV as stated in the documentation? Below is the exact message regarding this:

Error in private$check_score(self$score) : 
  Assertion on 'score' failed: Must be element of set {'partialling out'}, but is 'IV-type'.

**Comment by PhilippBach** [comment, marked as answer]
Hi @AdamBaumann ,

thanks for opening this discussion. Indeed, we are only supporting the `'partialling out'` score for PLIV with multiple instruments $Z$ at the moment. Right now, the implementations of more complex IV models is still experimental, so when you use such a model, please make sure you double check with a paper reference and verify the validity in a simulation. 

Basically, adapting the score in a way you proposed would require to adjust the lines below, I guess 

https://github.com/DoubleML/doubleml-for-r/blob/c9a67bced184ead56e7591ba4a369e95621fa0c6/R/double_ml_pliv.R#L693-L714

As far as I see, it should not be too much effort, so feel free to implement it yourself and open a pull request (see our [Contributing guidelines](https://github.com/DoubleML/doubleml-for-r/blob/main/CONTRIBUTING.md)).

Extending IV models is definitely on the list of the next features we want to include in DoubleML. We are currently working on some other topics, so we haven't completed the work on this yet.

**Reply by PhilippBach** [reply]
In case you need help or guidance when contributing to our package, feel free to contact us at anytime!
